### DISCUSSION TITLE
Build Failed: Execution failed for task ':model:generateProto'.

### DISCUSSION THREAD

**Discussion by MohitGupta121** [oppia/oppia-android#4526] [Q&A (Installation)] [answered]
<img width="1396" alt="build" src="https://user-images.githubusercontent.com/76530270/186341798-8919ef55-7116-41c0-acc8-ce7f6c8d7c41.png">

I am getting this error while building the project on Mac M1 air.

Android Studio Version:- Bumblebee

**Comment by rt4914** [comment, marked as answer]
@MohitGupta121 I have face similar issue today. 
For me this solution worked:

Replace if-else case in this [file](https://github.com/oppia/oppia-android/blob/ca39ac628bb5ebbefe22f3de59ecd3648188a37d/model/build.gradle#L7) with below code

```
    // Reference: https://github.com/grpc/grpc-java/issues/7690#issuecomment-760332746
    if (osdetector.os == "osx") {
      artifact = "com.google.protobuf:protoc:3.14.0:osx-x86_64"
    } else {
      artifact = "com.google.protobuf:protoc:3.14.0"
    }
```

**NOTE**:
- This code has not been tested across other os of mac so not sure if it fails for any other case or not.
- If you are applying this change make sure you do not commit this in your PR.

**Reply by MohitGupta121** [reply]
Thanks you @rt4914 it's works for me.

**Reply by Tejas-67** [reply]
hey @rt4914 , "If you are applying this change make sure you do not commit this in your PR." how do i do this?

**Reply by Vishwajith-Shettigar** [reply]
@Tejas-67 , just delete that code before committing. Use that code only while building project locally.
